Fix nullable fields handling with JSON source

VictorD-Veolia opened the pull request:
Fix an error in JSONPage that prevented HTTP sources to handle nullable fields correctly.

VictorD-Veolia committed:
fix

bdmogal commented:
@VictorD-Veolia do you mind sharing the exact issue (some sample response for which the plugin failed before the fix)? If you can add a test to demonstrate it, that would be even better.

bdmogal commented:
Also, since this repository does not have automatic builds set up yet, would it be possible to paste the output of `mvn clean package` with this change?

VictorD-Veolia committed:
add unit test for JSON Page

VictorD-Veolia commented:
Hi @bdmogal, I added a unit test and created an issue in JIRA : https://cdap.atlassian.net/browse/PLUGIN-805

I also added to this comment the output of the `mvn clean package` you asked.

[mvn_package.log](https://github.com/data-integrations/http/files/6974463/mvn_package.log)

fernst commented on pom.xml:
```diff
@@ -82,7 +82,7 @@
     <hadoop.version>2.3.0</hadoop.version>
     <httpcomponents.version>4.5.9</httpcomponents.version>
-    <hydrator.version>2.4.0-SNAPSHOT</hydrator.version>
+    <hydrator.version>2.4.0</hydrator.version>
```
We should bump to a release version in the develop branch

rmstar commented on src/test/java/io/cdap/plugin/http/source/common/pagination/page/JSONPageTest.java:
```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2019-2020 Cask Data, Inc.
```
2022

VictorD-Veolia committed:
clean

rmstar commented on pom.xml:
```diff
@@ -82,7 +82,7 @@
     <hadoop.version>2.3.0</hadoop.version>
     <httpcomponents.version>4.5.9</httpcomponents.version>
-    <hydrator.version>2.4.0-SNAPSHOT</hydrator.version>
+    <hydrator.version>2.10.0-SNAPSHOT</hydrator.version>
```
This should be set to a non-snapshot version

rmstar reviewed (approved):
please squash commits after addressing comment
